lighthouse-core/report/html/renderer/report-ui-features.js

```diff
@@ -391,8 +408,69 @@ class ReportUIFeatures {
    */
   onToolsButtonClick(e) {
     e.preventDefault();
-    this.toolsButton.classList.toggle('active');
-    this._document.addEventListener('keydown', this.onKeyDown);
+    e.stopImmediatePropagation();
+
+    if (this.toolsButton.classList.contains('active')) {
+      this.closeToolsDropdown();
+    } else {
+      this.openToolsDropdown();
+    }
+  }
+
+  /**
+   * Handler for tool button.
+   * @param {KeyboardEvent} e
+   */
+  onToolsButtonKeydown(e) {
+    switch (e.code) {
+      case 'ArrowUp':
+      case 'ArrowDown':
+      case 'Enter':
+      case ' ':
+        e.preventDefault();
+        this.openToolsDropdown();
+        break;
+      default:
+       // no op
+    }
+  }
+
+  /**
+   * Handler for tool dropdown.
+   * @param {KeyboardEvent} e
+   */
+  onToolsDropDownKeydown(e) {
+    const el = /** @type {?Element} */ (e.target);
+
+    if (!el) {
+      return;
+    }
+
+    let nextEl = el.nextElementSibling || this._dom.find(':first-child', this.toolsDropDown);
+    let prevEl = el.previousElementSibling || this._dom.find(':last-child', this.toolsDropDown);
+
+    if ((nextEl.getAttribute('disabled') !== null) ||
```

Comment:
maybe put these checks in a loop? today we don't have multiple disabled options in a row, but if we did this would break and be easy to forget.

```
let nextEl = el.nextElementSibling || this._dom.find(':first-child', this.toolsDropDown);
while (nextEl.getAttribute('disabled') !== null ||  window.getComputedStyle(nextEl).display === 'none') {
  nextEl = el.nextElementSibling || this._dom.find(':first-child', this.toolsDropDown); // just hope we never have everything disabled :P
}
```

Reply:
Done! In my new implementation I get the menu elements in an array and filter out the disabled and display = none.